### ISSUE TITLE
[java] CPD ignores excludes from ruleset

### ISSUE THREAD

**Issue by DevDavido** [pmd/pmd#5543]
**Affects PMD Version:** 3.26.0

**Description:**

Having this build configuration for the maven-pmd-plugin:
```xml
  ...
  <build>
    <plugins>
      <plugin>
        <artifactId>maven-pmd-plugin</artifactId>
        <version>3.26.0</version>
        <executions>
          <execution>
            <phase>compile</phase>
            <goals>
              <goal>check</goal>
              <goal>cpd-check</goal>
            </goals>
          </execution>
        </executions>
        <configuration>
          <failOnViolation>false</failOnViolation>
          <verbose>true</verbose>
          <rulesets>
            <ruleset>${pom.basedir}/ruleset.xml</ruleset>
          </rulesets>
        </configuration>
      </plugin>
  </build>
  ...
```

The ruleset.xml:

```xml
<ruleset name="Custom Ruleset" xmlns="http://pmd.sourceforge.net/ruleset/2.0.0"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://pmd.sourceforge.net/ruleset/2.0.0
                             http://pmd.sourceforge.net/ruleset_2_0_0.xsd">
  <exclude-pattern>.*/target/.*</exclude-pattern>
  ...
</ruleset>
```

**Steps to reproduce:**

Please provide detailed steps for how we can reproduce the bug.

1.  Run `mvn clean verify`
2.  Encounter CPD failures of actually excluded files:
```shell
[WARNING] CPD Failure: Found 22 lines of duplicated code at locations:
[WARNING]    /home/somedirs/project/target/generated-sources/openapi/src/gen/java/com/xxx/SomeClass.java line 140
[WARNING]    /home/somedirs/project/target/generated-sources/openapi/src/gen/java/com/xxx/SomeOtherClass.java line 146
```

**Running PMD through:** *Maven*

Eclipse Temurin 21, Ubuntu 22.04-LTS OS

**Comment by jsotuyod** [comment]
This is the correct behavior. Even if CPD and PMD are shipped together, they are independent tools, exposed under different ant tasks / maven targets / cli options. Ruleset definitions pertain exclusively to PMD check runs (ie: it defines applicable rules), while CPD has a separate specific configuration (ie: ignoring literals).

In Maven, the `cpd-check` goal has it's own configuration, which exposes it's own [`<excludes>`](https://maven.apache.org/plugins/maven-pmd-plugin/cpd-mojo.html#excludes) property for instance.
